astrospice/kernel.py

```diff
@@ -102,3 +103,77 @@ def coverage(self, body):
         body = Body(body)
         coverage = [t for t in spiceypy.spkcov(self._fname_str, body.id)]
         return Time(coverage, format='et').utc
+
+class MetaKernel(KernelBase):
+    """
+    A class for a single .tm kernel.
+
+    References
+    ----------
+    https://naif.jpl.nasa.gov/pub/naif/toolkit_docs/C/req/kernel.html#Additional%20Meta-kernel%20Specifications
+    """
+    _file_extension = '.tm'
+    
+    def __init__(self, fname):
+        """
+        Loading the metakernel will load all the kernels specified, if they exist.
+        """
+        self._fname = fname
+        if self.kernels_exist:
+            self.load_kernels()
+        else:
+            print("Kernels are not yet loaded")
+            
+    @property
+    def kernels(self):
+        """
+        The kernels specified by the metakernel
+
+        Returns
+        -------
+        list of kernel filenames specified in the metakernel
+        """
+        kernels = []
+        with open(self.fname, "r") as file:
+            look_for_kernels = False
+            for  line in file:
+                #split by whitespace
+                line_split = line.split()
+                
+                # now find the ) bracket by itself, this is the end of kernels to load
+                if len(line_split) >= 1 and line_split[0] == ')':
+                    look_for_kernels = False
+                    break
+                
+                if look_for_kernels and len(line_split) > 0:
+                    # find the filename for the kernel. 
+                    # The slicing removes the $KERNEL and the final '
+                    kernel_fname = line_split[0][9:-1]
+                    #do not add empty lines
+                    if kernel_fname != '':
+                        #slice removes the first \
+                        kernels.append(kernel_fname[1:])
+                
+                if len(line_split) > 1 and line_split[0] == 'KERNELS_TO_LOAD':
+                    look_for_kernels = True
+        
+        return kernels
+    
+    def load_kernels(self):
+        """
+        Loads the kernels specified by the metakernel
+        """
+        for kernel in self.kernels:
+            KernelBase(self.fname.parent / kernel)
```

Comment:
```suggestion
            Kernel(self.fname.parent / kernel)
```
This should automatically select the right class for each kernel.

Reply:
I left it as `KernelBase` as that furnishes all the files. If this is `Kernel` it will raise an error that this not a supported file type in astrospice.

Reply:
👍 I'll fix that in another PR